tests/test_when.py

```diff
@@ -485,6 +485,7 @@ class _Test(m.Circuit):
     assert check_gold(__file__, f"{basename}.mlir")
 
 
+@pytest.mark.skip("There's a bug in this test")
```

Comment:
nit: let's make a more descriptive skip message, maybe `"Test does not pass due to when issue (#1156)"`